### PR TITLE
Defenders can now parry thrown attacks with tail-abilities

### PR DESCRIPTION
**Pull request by DeltaFire15**
## About The Pull Request
Title.
Tail-attacks (Swipe and Centrifugal Force) now have a small parry window of about the animation time. During that time, if they are hit by anything thrown (nade, throwing knife, etc -- not airburst nades because those are bullets - curse those), they will deflect them back towards their source.
Example:
nade_parry
(Yes I know nade launchers have no warning icon period pre-firing, I had to code myself a firing test so I didn't have to conscript friends)

This is done via a component only Defender has / interacts with for now, although this should be expandable to other xenos if wanted, or completely different things with slight tweaks (machetes / harvesters parrying facehuggers? Yeah sure, doable with a bit more work)

Is this a joke PR? Uhhhhhhhhh who knows, it works pretty well and it was surprisingly fun to code, however touching legacy throw code was a bit tedious.

## Why It's Good For The Game
This is a fun mechanic fitting for Defender's defense, which as opposed to the bullet deflection PR  should have lower impact aswell as require skill to pull off.

## Changelog
:cl:
add: Defenders can use their tail abilities (Swipe, Centrifugal Force) to parry throws, if timed right.
/:cl:

### CODE REVIEW

**Comment by lKiseki** [comment]
Holy based.

**Comment by Lumipharon** [comment]
Super cool feature man.
Also we could totally code in base ball or something with this.

**Comment by ilikesaying** [comment]
Chad defender batting the Willy Pete back towards the muhreens?

**Comment by vilereaver41** [comment]
Good luck timing that lol

**Comment by MalricB** [comment]
ayo pog

**Comment by ThelastArchiver** [comment]
this sounds perfectly balanced as all things shouls be

File: code/modules/mob/living/carbon/xenomorph/castes/defender/abilities_defender.dm [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
 	X.add_filter("defender_tail_sweep", 2, gauss_blur_filter(1)) //Add cool SFX
 	X.spin(4, 1)
+	SEND_SIGNAL(owner, COMSIG_PARRY_TRIGGER, 0.6 SECONDS)
```
[the remark this answers is not in the document]
Comment (DeltaFire15) [reply code/modules/mob/living/carbon/xenomorph/castes/defender/abilities_defender.dm:32]: A bunch of signals have some different trigger locations and this seems cleaner than having a /mob proc that sends the signal, however, sure.

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -545,6 +563,10 @@
 	//done throwing, either because it hit something or it finished moving
 	if(!originally_dir_locked)
 		flags_atom &= ~DIRLOCK
+	if(parrier)
+		sleep(1)	//Very small sleep so it doesn't look like it just teleported back to you if you were very close due to how throw code works.
+		throw_at((thrower && thrower != src) ? thrower : throw_source, range, max(1, speed/2), parrier, spin, flying)
```
[the remark this answers is not in the document]
Comment (DeltaFire15) [reply code/game/atoms_movable.dm:568]: I didn't know that existed, so, sure.

**Comment by Pariah919** [comment]
funny? yes.
good mechanic? probably not, but hey what's the worst that could happen

**Comment by Losenis** [comment]
How does this work with the primo ability?

Also can this parry OB

**Comment by DeltaFire15** [comment]
> How does this work with the primo ability?
Primordial makes you swipe repeatedly and every swipe triggers parrying. Basically, if someone is dumb enough to throw a nade at a beyblade defender, they'll reap the consequences.

> Also can this parry OB
OBs are not a thrown thing, so no.

**Review by TiviPlus — APPROVED** [review]

**Comment by Naaanii** [comment]
Much better alternative to deflect.

**Comment by Losenis** [comment]
>OBs are not a thrown thing, so no.

Marines throw hard with them, what do you mean

**Pull request merged**